ucu-instagram/src/app/nav/nav.component.ts

```diff
@@ -10,28 +9,28 @@ import { UploadComponent } from '../upload/upload.component';
   styleUrls: ['./nav.component.css']
 })
 export class NavComponent {
-@Input() selected! : "discover" | "search"  | "upload" | "chat" | "profile" ;
 
-constructor(private router : Router, private dialog : MatDialog){
+  selected: "discover" | "search" | "upload" | "chat" | "profile" = "discover";
 
-}
+  constructor(private router: Router, private dialog: MatDialog) {
+  }
 
-  navigate(newView : "discover" | "search"  | "upload" | "chat" | "profile") {
+  navigate(newView: "discover" | "search" | "upload" | "chat" | "profile") {
     this.dialog.closeAll();
-    setTimeout(()=> {
-      this.selected = newView ;
+    this.selected = newView;
+    if (newView === "discover" || newView === "search" || newView === "profile") {
```

Comment:
What's the purpose of this if statement? It looks like you should restrict the input type to only those values

Reply:
the selected field should always change to the new value, but we don't want to navigate to the specified route always, for instance there's no 'chat' route, as it is just a dialog, but there is a 'discover' route.